repose-aggregator/components/filters/uri-stripper-filter/src/test/scala/org/openrepose/filters/uristripper/UriStripperFilterRequestTest.scala

```diff
@@ -0,0 +1,1028 @@
+            |            <xml>
+            |                <namespace name="foo" url="bar"/>
+            |                <xpath>/root/foo:service/foo:link</xpath>
+            |            </xml>
+            |        </request>
+            |    </link-resource>
+            |</uri-stripper>
+          """.stripMargin
+
+        val requestBody =
+          """<?xml version="1.0" encoding="UTF-8"?>
+            |<root xmlns:foo="bar">
+            |  <foo:service>
+            |    <foo:link>http://example.com/v1/12345/foo</foo:link>
+            |  </foo:service>
+            |</root>
+          """.stripMargin
+
+        filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+        request.setRequestURI("/v1/12345/foo")
+        request.setContent(requestBody.getBytes(CHARSET_UTF8))
+        request.setContentType(MimeType.APPLICATION_XML.toString)
+
+        filter.doFilter(request, response, filterChain)
+
+        getPostFilterRequestBody contains "<foo:link>http://example.com/v1/foo</foo:link>"
+      }
+
+      it("should fail if the namespaces used in the request are not specified in the config (fail)") {
+        val config =
+          """<?xml version="1.0" encoding="UTF-8"?>
+            |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+            |    <link-resource uri-path-regex=".*">
+            |        <request>
+            |            <xml>
+            |                <namespace name="foo" url="bar"/>
+            |                <xpath link-mismatch-action="fail">/service/link</xpath>
+            |            </xml>
+            |        </request>
+            |    </link-resource>
+            |</uri-stripper>
+          """.stripMargin
+
+        val requestBody =
+          """<?xml version="1.0" encoding="UTF-8"?>
+            |<badnamespace:service xmlns:badnamespace="bar">
+            |  <badnamespace:link>http://example.com/v1/12345/foo</badnamespace:link>
+            |</badnamespace:service>
+          """.stripMargin
+
+        filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+        request.setRequestURI("/v1/12345/foo")
+        request.setContent(requestBody.getBytes(CHARSET_UTF8))
+        request.setContentType(MimeType.APPLICATION_XML.toString)
+
+        filter.doFilter(request, response, filterChain)
+
+        response.getStatus shouldEqual HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+        response.getContentLength shouldEqual 0
+      }
+    }
+  }
+
+  def getPostFilterRequestBody: String = {
+    val requestCaptor = ArgumentCaptor.forClass(classOf[HttpServletRequestWrapper])
+    verify(filterChain).doFilter(requestCaptor.capture(), any(classOf[ServletResponse]))
+    scala.io.Source.fromInputStream(requestCaptor.getValue.getInputStream)(scala.io.Codec.UTF8).mkString
+  }
+}
+
+object UriStripperFilterRequestTest {
+  val CHARSET_UTF8 = Charset.forName("UTF-8")
+}
```

Comment:
Yep. The existing tests were very through in their coverage, so duplicating them for the request side just made sense.